File: src/main/java/com/freya02/botcommands/prefixed/annotation/Hidden.java
package com.freya02.botcommands.prefixed.annotation;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/** Use this annotation if you want the command to be hidden from everyone but the owners, <b>the 'help' command will not display it</b>
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.TYPE})
public @interface Hidden {}

File: BotCommands-commands/text/src/main/kotlin/io/github/freya022/botcommands/api/commands/text/annotations/Hidden.kt
package io.github.freya022.botcommands.api.commands.text.annotations

import io.github.freya022.botcommands.api.commands.text.builder.TextCommandBuilder
import io.github.freya022.botcommands.api.core.BotOwners

/**
 * Hides a command and its subcommands from help content and execution,
 * except for [bot owners][BotOwners].
 *
 * ### Annotation scope
 * This annotation applies to the command itself, not just on a single variation,
 * in other words, it applies to all commands sharing the same path.
 *
 * @see TextCommandBuilder.hidden DSL equivalent
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.CLASS, AnnotationTarget.FUNCTION)
annotation class Hidden  
